app/assets/javascripts/components/samples.jsx

```diff
@@ -98,6 +102,39 @@ class Samples extends React.Component {
     });
   }
 
+  displayReportProgress(res) {
+      $('.download-progress')
+      .html(`${res.data.status_display}`)
+      .css('display', 'block');
+
+      setTimeout(() => {
+        this.checkReportDownload()
+      }, 2000)
```

Comment:
does the polling stop after the report generation reaches a terminal state?

every 2 seconds seems real aggressive;   perhaps start at 5 seconds, and add 5 on each wakeup

Reply:
Yes, it stops after it actually gets the file. Ok, will increase